helix-core/src/test/java/org/apache/helix/integration/task/TestForceDeleteWorkflow.java

```diff
@@ -0,0 +1,337 @@
+      return (wCtx1.getWorkflowState() == TaskState.STOPPED);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowStopped);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed created for this
+    // workflow
+    Assert.assertNotNull(_driver.getWorkflowConfig(workflowName));
+    Assert.assertNotNull(_driver.getWorkflowContext(workflowName));
+    Assert.assertNotNull(admin.getResourceIdealState(CLUSTER_NAME, workflowName));
+
+    // Disable the cluster before Force Delete
+    admin.enableCluster(CLUSTER_NAME, false);
+    Thread.sleep(Disable_Delay);
+
+    _driver.delete(workflowName, true);
+    Thread.sleep(DELETE_DELAY);
+
+    // Enable the cluster before Force Delete
+    admin.enableCluster(CLUSTER_NAME, true);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed deleted for this
+    // workflow
+    Assert.assertNull(_driver.getWorkflowConfig(workflowName));
+    Assert.assertNull(_driver.getWorkflowContext(workflowName));
+    Assert.assertNull(admin.getResourceIdealState(CLUSTER_NAME, workflowName));
+  }
+
+  @Test
+  public void testDeleteNotStartedWorkflowForcefully() throws Exception {
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = new Workflow.Builder(workflowName);
+
+    //Workflow Schematic:
+    //               JOB0
+    //                /\
+    //               /  \
+    //             JOB1 JOB2
+
+    JobConfig.Builder jobBuilder0 = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "10000"));
+
+    JobConfig.Builder jobBuilder1 = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "10000"));
+
+    JobConfig.Builder jobBuilder2 = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "10000"));
+
+    // Execution delay if set to be large number in order to have enough time to change the status
+    // of the workflow to not started.
+    builder.addParentChildDependency("JOB0", "JOB1");
+    builder.addParentChildDependency("JOB0", "JOB2");
+    builder.addJob("JOB0", jobBuilder0.setExecutionDelay(10000000000L));
+    builder.addJob("JOB1", jobBuilder1);
+    builder.addJob("JOB2", jobBuilder2);
+
+    _driver.start(builder.build());
+    // Wait until workflow is created.
+    boolean isWorkflowCreated = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      return (wCtx1 != null);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowCreated);
+
+    // Change the workflow state to Not-Started and set its start time of the workflow.
```

Comment:
Workflow may have started but you are just overwriting the state to be NOT_STARTED?

Reply:
I am deleting this test for now.